### PR TITLE
Strip code cell warnings

### PR DESCRIPTION
**Pull request by GeorgianaElena**
This PR strips cell warnings when voila is run without the ``--debug`` flag, similar with how it's done for errors.

I'm not sure if it's enough to fix the related issues (https://github.com/voila-dashboards/voila/issues/381, https://github.com/voila-dashboards/voila/issues/252) or if there's a better approach to this :eyes: .

### CODE REVIEW

**Comment by martinRenou** [comment]
I guess it should be enough for fixing the issues you mentioned

File: voila/execute.py [hunk 1]
```diff
@@ -35,13 +35,30 @@ def strip_code_cell_errors(cell):
...
+    # There is no 'outputs' key for markdown cells
+    if 'outputs' not in cell:
+        return cell
+
+    outputs = cell['outputs']
+
+    warning_outputs = [output for output in outputs if output['output_type'] == 'stream' and output['name'] == 'stderr']
+
+    for output in warning_outputs:
+        output['name'] = ''
```
Comment (martinRenou) [inline voila/execute.py]: Shouldn't you set the "text" key to an empty string as well?

Comment (GeorgianaElena) [reply voila/execute.py]: I thought about this, but by setting the "name" key to an empty string, the warning message doesn't get printed to any output stream and I thought it won't make a difference if I'd change/emptied the text. But I'm not sure I got the logic right :grimacing:. Am I making any sense?

Comment (martinRenou) [reply voila/execute.py]: Yes! Maybe the front-end does not show anything by default if the name is empty, I did not check what the logic is. But just to be sure I would set the text to empty string too...

Thinking about it, it might be even better to remove those outputs from the cell outputs list. What do you think?

Comment (GeorgianaElena) [reply voila/execute.py]: I agree it'll be better to remove the warning outputs entirely.

Comment (GeorgianaElena) [reply voila/execute.py]: Updated the PR! Thanks a lot for the feedback :)

File: voila/execute.py [hunk 2]
```diff
@@ -35,13 +35,30 @@ def strip_code_cell_errors(cell):
...
+    # There is no 'outputs' key for markdown cells
+    if 'outputs' not in cell:
+        return cell
+
+    outputs = cell['outputs']
+
+    def is_warning(output):
+        return output['output_type'] == 'stream' and output['name'] == 'stderr'
+
+    cell['outputs'] = [output for output in outputs if not is_warning(output)]
```
[the remark this answers is not in the document]
Comment (martinRenou) [reply voila/execute.py]: I don't want to block your PR for this kind of details so this can be done later if you want this fix soon

[the remark this answers is not in the document]
Comment (GeorgianaElena) [reply voila/execute.py]: No worries, there's no hurry, I'm just excited for my first PR to Voila :heart:

[the remark this answers is not in the document]
Comment (martinRenou) [reply voila/execute.py]: Thanks a lot for your PR :) !!

**Comment by martinRenou** [comment]
@SylvainCorlay @maartenbreddels @jtpio

**Comment by maartenbreddels** [comment]
Hi Georgiana,

thanks for you first time contribution! I recognize that this is something you often want, but I am not sure this should be default behavior. Are there examples of where the output to stderr is actually something you would like to show to a user? Maybe not, and we should just merge this.

Anyone else with some input?

cheers,

Maarten

**Comment by martinRenou** [comment]
> I recognize that this is something you often want, but I am not sure this should be default behavior

The same way we strip errors by default, I suppose we should strip warnings by default (they could leak some code, information...).
Note that when running with `--debug` you will see warning with this PR.

**Comment by maartenbreddels** [comment]
But with an exception, you get feedback that something happened, now we basically send stderr to /dev/null. I am not sure that it should go under the same flag, because now you cannot hide exceptions, but show stderr. The workaround for this could be to use an output widget btw.

**Comment by martinRenou** [comment]
So you mean you would rather see something like:

```python
warnings = [
    output for output in cell['outputs']
    if output['output_type'] == 'stream' and output['name'] == 'stderr'
]

for warning in warnings:
    warning['name'] = 'warning'
    warning['text'] = 'There was a warning in cell {}, run Voila with --debug if you want to see it.'
```

?

**Comment by jtpio** [comment]
> The same way we strip errors by default, I suppose we should strip warnings by default (they could leak some code, information...).

I also think it makes sense to strip warnings by default.

**Comment by martinRenou** [comment]
@maartenbreddels do you mind having another look a this if we're making a release? Do you think we should have an equivalent logic as with error stripping (see my previous comment)?

**Comment by maartenbreddels** [comment]
Ok, I think indeed the default behaviour should be not to show the error output, since it is stderr or warnings, you don't want to see it, and it may leak information. What I am unsure about, is that `--debug` should be the only way to turn this on (since it will also show exception). But if we add extra flags for this later, this won't be a breaking change.

TLDR; let's merge this as it, I prefer to squash and merge it.

**Comment by jtpio** [comment]
> What I am unsure about, is that `--debug` should be the only way to turn this on (since it will also show exception).

That could be tweaked with `--Voila.log_level` then?

**Comment by maartenbreddels** [comment]
> That could be tweaked with `--Voila.log_level` then?

I think that's more for the internal logging. That may be a per template settting actually, your debug template might want to show that by default? Or should it be a per app setting?

**Comment by maartenbreddels** [comment]
Linux is green, OSX takes ages, I'm going to merge this. Thanks for your first contribution @GeorgianaElena

**Comment by SylvainCorlay** [comment]
Thanks @GeorgianaElena ! This should be in the release later today.

**Comment by jtpio** [comment]
> I think that's more for the internal logging. That may be a per template settting actually, your debug template might want to show that by default? Or should it be a per app setting?

Well for now `--debug` is just an alias for `--Voila.log_level=logging.DEBUG`.
